hoomd/mpcd/CollisionMethod.cc

```diff
@@ -48,15 +78,945 @@ void mpcd::CollisionMethod::collide(uint64_t timestep)
+                    // add to accumulating center of mass
+                    center_of_mass.x += local_pos.x * mass;
+                    center_of_mass.y += local_pos.y * mass;
+                    center_of_mass.z += local_pos.z * mass;
+                    mass_sum += mass;
+                    }
+                // check if center of mass in body frame is (0, 0, 0)
+                bool in_tol = (center_of_mass.x >= -tol && center_of_mass.x <= tol)
+                              && (center_of_mass.y >= -tol && center_of_mass.y <= tol)
+                              && (center_of_mass.z >= -tol && center_of_mass.z <= tol);
+
+                invalid_center_of_mass = invalid_center_of_mass || !in_tol;
+                // check if center of mass is the same as the sum of masses
+                Scalar mass_diff = center_mass - mass_sum;
+                bool sum_in_tol = mass_diff >= -1.0 * tol && mass_diff <= tol;
+                invalid_mass_sum = invalid_mass_sum || !sum_in_tol;
+                }
+            }
+
+#ifdef ENABLE_MPI
+        if (m_exec_conf->getNRanks() > 1)
+            {
+            MPI_Allreduce(MPI_IN_PLACE,
+                          &invalid_mass,
+                          1,
+                          MPI_CXX_BOOL,
+                          MPI_LOR,
+                          m_exec_conf->getMPICommunicator());
+            MPI_Allreduce(MPI_IN_PLACE,
+                          &needs_thermostat,
+                          1,
+                          MPI_CXX_BOOL,
+                          MPI_LOR,
+                          m_exec_conf->getMPICommunicator());
+            MPI_Allreduce(MPI_IN_PLACE,
+                          &central_interacting,
+                          1,
+                          MPI_CXX_BOOL,
+                          MPI_LOR,
+                          m_exec_conf->getMPICommunicator());
+            MPI_Allreduce(MPI_IN_PLACE,
+                          &invalid_center_of_mass,
+                          1,
+                          MPI_CXX_BOOL,
+                          MPI_LOR,
+                          m_exec_conf->getMPICommunicator());
+            MPI_Allreduce(MPI_IN_PLACE,
+                          &invalid_mass_sum,
+                          1,
+                          MPI_CXX_BOOL,
+                          MPI_LOR,
+                          m_exec_conf->getMPICommunicator());
+            }
+#endif // ENABLE_MPI
+
+        if (needs_thermostat)
+            {
+            requireTemperature();
+            }
+        if (invalid_mass)
+            {
+            throw std::runtime_error("Some particles have a mass <= 0.");
+            }
+        if (central_interacting)
+            {
+            m_exec_conf->msg->warning() << "Central particle of rigid body included in "
+                                           "MPCD collision. Check if this is intentional."
+                                        << std::endl;
+            }
```

Comment:
I would put the warning first, before you do the more severe errors.
```suggestion
        if (central_interacting)
            {
            m_exec_conf->msg->warning() << "Central particle of rigid body included in "
                                           "MPCD collision. Check if this is intentional."
                                        << std::endl;
            }
        if (invalid_mass)
            {
            throw std::runtime_error("Some particles have a mass <= 0.");
            }
```